engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1509,41 +1518,51 @@ private static boolean serializeAllTable(
 
         final String[] columnSources = table.getDefinition().getColumnNamesArray();
 
-        try (final SharedContext sharedContext =
-                (columnSources.length > 1) ? SharedContext.makeSharedContext() : null) {
-            for (int ii = 0; ii < columnSources.length; ++ii) {
-                if (concurrentAttemptInconsistent()) {
-                    if (log.isDebugEnabled()) {
-                        final LogEntry logEntry = log.debug().append(System.identityHashCode(logIdentityObject))
-                                .append(" Bad snapshot before column ").append(ii);
-                        appendConcurrentAttemptClockInfo(logEntry);
-                        logEntry.endl();
-                    }
-                    return false;
+        final ExecutionContext executionContext = ExecutionContext.getContext();
+        final JobScheduler jobScheduler;
+        // TODO Check with Ryan if I should keep this check
+        // if (executionContext.getOperationInitializer() instanceof PoisonedOperationInitializer) {
+        // jobScheduler = new ImmediateJobScheduler();
+        // }
+        if (ENABLE_PARALLEL_SNAPSHOT && executionContext.getOperationInitializer().canParallelize() &&
+                (snapshot.rowsIncluded.size() > MINIMUM_PARALLEL_SNAPSHOT_ROWS ||
+                        !allColumnSourcesInMemory(table, columnSources, columnsToSerialize))) {
+            // We can parallelize the snapshot
+            jobScheduler = new OperationInitializerJobScheduler();
+        } else {
+            jobScheduler = new ImmediateJobScheduler();
+        }
+
+        final CompletableFuture<Boolean> waitForResult = new CompletableFuture<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        jobScheduler.iterateParallel(
+                executionContext,
+                logOutput -> logOutput.append("serializeAllTable"),
+                JobScheduler.DEFAULT_CONTEXT_FACTORY,
+                0, columnSources.length,
+                (context, colIdx, nestedErrorConsumer) -> serializeColumn(columnSources, colIdx, usePrev, snapshot,
+                        table, logIdentityObject, columnsToSerialize),
+                () -> waitForResult.complete(Boolean.TRUE),
+                err -> {
+                    exception.set(err);
+                    waitForResult.complete(Boolean.FALSE);
+                });
+        try {
+            final boolean ret = waitForResult.get();
```

Comment:
You could allow `SnapshotInconsistentException` to propagate wrapped in an `ExecutionException`.